### PR TITLE
Added TimeZone and missing TimeZoneInfo members

### PR DESCRIPTION
**Pull request by hann013**
Needed in order to complete [corefx #10074](https://github.com/dotnet/corefx/issues/10074)

### CODE REVIEW

File: src/mscorlib/model.xml [hunk 1]
```diff
@@ -7425,16 +7425,29 @@
       <Member MemberType="Property" Name="TotalMinutes" />
       <Member MemberType="Property" Name="TotalSeconds" />
     </Type>
+    <Type Name="System.TimeZone">
```
Comment (jasonwilliams200OK) [inline src/mscorlib/model.xml]: Apparently, there is no `public abstract class TimeZone` in CoreCLR as defined by full framework.

**Comment by weshaggard** [comment]
@tarekgh @ellismg any suggestions on TimeZone? Should we make it standalone or put it in here?

**Comment by tarekgh** [comment]
@weshaggard I am not expecting to touch the TimeZone code anymore so I don't think it is worth to move it to somewhere else.

**Comment by tarekgh** [comment]
@hann013 do we already have the Linux/OSX implementation for TimeZone?

**Comment by hann013** [comment]
@tarekgh We have the mono implementation.

File: src/mscorlib/model.xml [hunk 2]
```diff
@@ -7495,8 +7495,7 @@
       <Member Name="get_DaylightTransitionEnd" />
       <!-- CreateAdjustmentRule doesn't have a public contract, but tests and other callers can use Reflection to invoke it, so ensure it is left in mscorlib. -->
       <Member Status="ApiRoot" Name="CreateAdjustmentRule(System.DateTime,System.DateTime,System.TimeSpan,System.TimeZoneInfo+TransitionTime,System.TimeZoneInfo+TransitionTime)" />
-      <Member Name="Equals(System.Object)" />
```
Comment (weshaggard) [inline src/mscorlib/model.xml]: Why did you remove Equals(System.Object)?

File: src/mscorlib/src/System/CurrentSystemTimeZone.cs [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+                    time = time.AddDays(-delta);
+                }
+            }
+            return (time);
+        }
+
+        [System.Security.SecurityCritical]  // auto-generated
+        [ResourceExposure(ResourceScope.None)]
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal extern static int nativeGetTimeZoneMinuteOffset();
```
Comment (ellismg) [inline src/mscorlib/src/System/CurrentSystemTimeZone.cs]: I don't expect that these internal calls into the run-time are going to work on Linux/OSX.  Most of the globalization support is not present in the run-time in these cases.

We probably need to change the implementation of this file to not use these methods at least for OSX and Linux.

Comment (tarekgh) [inline src/mscorlib/src/System/CurrentSystemTimeZone.cs]: @ellismg I already asked @hann013 regarding Linux based implementation and he mentioned will pick Mono implementation.

Comment (justinvp) [inline src/mscorlib/src/System/CurrentSystemTimeZone.cs]: Could `TimeZone` be implemented (compatibly) by calling `TimeZoneInfo`?

File: src/mscorlib/model.xml (hunk 2 above)
Comment (hann013) [inline src/mscorlib/model.xml]: I realized that there isn't an implementation of `Equals(System.Object)` within `TimeZoneInfo.AdjustmentRule`.

Comment (tarekgh) [inline src/mscorlib/model.xml]: I think you need to keep 

<Member Name="Equals(System.Object)" />

**Comment by hann013** [comment]
Fixed by @joperezr in #6820
